app/src/main/java/org/mifos/selfserviceapp/presenters/UserDetailsPresenter.java

```diff
@@ -0,0 +1,127 @@
+
+    @Inject
+    public UserDetailsPresenter(@ApplicationContext Context context, DataManager dataManager) {
+        super(context);
+        this.dataManager = dataManager;
+        subscriptions = new CompositeSubscription();
+    }
+
+    @Override
+    public void attachView(UserDetailsView mvpView) {
+        super.attachView(mvpView);
+    }
+
+    @Override
+    public void detachView() {
+        super.detachView();
+        subscriptions.unsubscribe();
+    }
+
+    public void getUserDetails() {
+        checkViewAttached();
+        getMvpView().showProgress();
+        subscriptions.add(dataManager.getCurrentClient()
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribeOn(Schedulers.io())
+                .subscribe(new Subscriber<Client>() {
+                    @Override
+                    public void onCompleted() {
+
+                    }
+
+                    @Override
+                    public void onError(Throwable e) {
+                        getMvpView().showError(context.getString(R.string.error_fetching_client));
+                        getMvpView().hideProgress();
+                    }
+
+                    @Override
+                    public void onNext(Client client) {
+                        getMvpView().hideProgress();
+                        if (client != null) {
+                            getMvpView().showUserDetails(client);
+                        } else {
+                            getMvpView().showError(context
+                                    .getString(R.string.error_client_not_found));
+                        }
+                        getMvpView().hideProgress();
+                    }
+                })
+        );
+    }
+
+    public void getUserImage() {
+        checkViewAttached();
+        subscriptions.add(dataManager.getClientImage()
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribeOn(Schedulers.io())
+                .subscribe(new Subscriber<ResponseBody>() {
+                    @Override
+                    public void onCompleted() {
+
+                    }
+
+                    @Override
+                    public void onError(Throwable e) {
+
+                    }
+
+                    @Override
+                    public void onNext(ResponseBody response) {
+                        try {
+                            final String encodedString = response.string();
+
+                            //removing 'data:image/jpg;base64' from the response
+                            //the response is of the form of 'data:image/jpg;base64, .....'
+                            final String pureBase64Encoded =
+                                    encodedString.substring(encodedString.indexOf(',') + 1);
+
+                            final byte[] decodedBytes =
+                                    Base64.decode(pureBase64Encoded, Base64.DEFAULT);
+
+                            Bitmap decodedBitmap =
```

Comment:
I am not very familiar with RxJava concepts. So I have this doubt about, on which thread will this code be called?

I see that you are observing on ```AndroidSchedulers.mainThread()```, I don't think it is a good idea to decode the bitmap on the main thread.

Please throw some light on this, thanks!